### PR TITLE
Use CancellationToken in AvaloniaNameGenerator

### PR DESCRIPTION
**Pull request by jankrib**
## What does the pull request do?
I was experiencing severe performance issues in visual studio with a larger project with many axaml files. The better solution here is probably to implement the newer `IIncrementalGenerator` interface, but I noticed that the generation was not using the CancellationToken. This made a huge difference for me.

## What is the current behavior?
It keeps generating all the xaml files even if the operation is cancelled

## What is the updated/expected behavior with this PR?
The generation is cancelled if the CancellationToken reports so.

## How was the solution implemented (if it's not obvious)?
I tried to churn as little as possible, but had to change the return type from IReadOnlyList to IEnumerable to get the lazy behavior wanted by the foreach loop.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037391-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by timunie** [comment]
I think incremental generator was planned, but not available back then. Don't know how easy it would be to implement that interface at this stage though.

**Comment by jankrib** [comment]
I looked into it. It is pretty trivial but requires refactoring. I gave it a quick go, but the way options are made from the old context made me try this simple fix instead. Using it now it doesn't feel slow and the project is quite large.

File: src/tools/Avalonia.Generators/NameGenerator/AvaloniaNameSourceGenerator.cs [hunk 1]
```diff
@@ -22,8 +22,16 @@ public void Execute(GeneratorExecutionContext context)
...
+            {
+                if(context.CancellationToken.IsCancellationRequested)
+                {
+                    break;
+                }
+
+                context.AddSource(fileName, content);
+            }
         }
         catch (Exception exception)
```
Comment (maxkatz6) [inline src/tools/Avalonia.Generators/NameGenerator/AvaloniaNameSourceGenerator.cs:36]: OperactionCancelledException can be handled with no-op here. But it depends if file.GetText throws on cancellation or not

Comment (jankrib) [reply src/tools/Avalonia.Generators/NameGenerator/AvaloniaNameSourceGenerator.cs:39]: Agreed. Fixed now

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037432-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Pull request merged**